content/blog/what-are-webhooks.md

```diff
@@ -0,0 +1,196 @@
+
+In the past, before the advent of webhooks, developers relied on a variety of methods to transfer data between systems. Let's explore some of the notable techniques and the challenges they pose.
+
+### Polling
+
+One standard method was polling, where an application repeatedly queries another application for new data. For example, an application might periodically check an email server for new messages. This approach had significant drawbacks, including the waste of resources as applications had to continually send requests, even when no new data was available. It also resulted in delays between data updates, as the system had to wait for the next polling interval.
+
+### Messaging Queues
+
+Some systems use messaging queues to enable communication between applications. In this approach, applications would send messages to a central queue, and other applications would read these messages and process them accordingly. While messaging queues improved data transfer efficiency compared to polling, they often required additional infrastructure and complex configuration. Setting up and maintaining a messaging queue system could be daunting, particularly for smaller organizations or developers with limited resources.
+
+### Remote Procedure Calls (RPC)
+
+RPC is a synchronous communication mechanism, but it can also be used to achieve asynchronous communication by implementing callback functions or using futures and promises. With callbacks, the client initiates a request and provides a callback function that the server invokes when the result is ready. [Futures and Promises](https://en.wikipedia.org/wiki/Futures_and_promises) allow the client to request a result and receive a future object immediately. The client can then periodically check the future object for completion or attach callbacks to handle the result when it becomes available. However, RPC is designed to handle synchronous communication, and forcing it to handle asynchronous scenarios might feel unnatural and can lead to scalability and flexibility issues.
+
+### Scheduled Jobs
+
+Scheduled jobs were another way to transfer data between applications. These jobs would run at predefined intervals and perform specific tasks, such as fetching data from one application and pushing it to another. However, scheduled jobs suffered the same drawbacks as polling, such as delays in data updates and wasted resources when no new data was available.
+
+### Polling Optimization
+
+To reduce the impact of frequent polling, developers implemented techniques like long-polling. These methods involved keeping the connection open until new data was available, minimizing the number of requests made to the server. While this improved real-time updates to some extent, it still had drawbacks, such as increased server load and higher latency.
+
+### Comet Servers
```

Comment:
@multi-Adams It will also be helpful to mention Comet Servers were invented for server-to-browser communication and not server-to-server communication.

Reply:
It'll be great to add links where necessary, for example -- The coining of [Comet](https://infrequently.org/2006/03/comet-low-latency-data-for-the-browser/) Servers.

Reply:
Noted. 

I'll make the necessary changes.

Reply:
done, please review